app/code/community/Apruve/ApruvePayment/controllers/WebhookController.php

```diff
@@ -25,44 +25,44 @@ public function updateOrderStatusAction()
         $hash = $this->_getHashedQueryString();
 
         // if the hash doesn't match the data sent by Apruve terminate the code
-        if (!isset($_GET[$hash])) {
+        if (! isset($_GET[$hash])) {
             header("HTTP/1.1 404 Not Found");
             exit;
         }
 
         // if the hash matches the data sent by Apruve move forward with the appropriate process
         $input = file_get_contents('php://input');
-        $data = json_decode($input);
+        $data  = json_decode($input);
 
         Mage::helper('apruvepayment')->logException($data);
         try {
-            $event = $data->event;
+            $event  = $data->event;
             $entity = $data->entity;
 
             // check the event triggered in Apruve to call appropriate action in Magento
-            if ($event == 'invoice.closed') {
+            if ($event == 'invoice.funded') {
```

Comment:
We need to handle both invoice.funded and invoice.closed. If an invoice is self-financed, we will never send an invoice.funded webhook for it because there will be no creditor to fund it. The invoice should get closed out whenever either webhook is received. If we get either webhook after the invoice is already closed out, we should ignore it.